### PR TITLE
feat(esl-popup): internal ESLPopup implementation updated to use ESLEventListeners; usage of cached properties reduced

### PR DESCRIPTION
**Pull request by ala-n**
Closes: #2009

### CODE REVIEW

File: src/modules/esl-popup/core/esl-popup.ts [hunk 1]
```diff
@@ -186,15 +172,16 @@ export class ESLPopup extends ESLToggleable {
...
     this.activator = params.activator;
-    if (this.open) this.afterOnShow();
-    if (this.open && !params.force) return false; // the show task will be forced to run so the next steps are unnecessary
+    if (this.open) {
+      this.afterOnHide();
+      this.afterOnShow();
+    }
+
+    if (!params.force && this.open) return false;
+    if (!params.silent && !this.$$fire(this.BEFORE_SHOW_EVENT, {detail: {params}})) return false;
```
Comment (ala-n) [inline src/modules/esl-popup/core/esl-popup.ts:184]:
```suggestion
```

File: src/modules/esl-popup/core/esl-popup.ts [hunk 2]
```diff
@@ -186,15 +172,15 @@ export class ESLPopup extends ESLToggleable {
...
-   * @returns false if the show task should be canceled
-   */
+  /** Runs additional actions on show popup request */
   protected override onBeforeShow(params: ESLToggleableActionParams): boolean | void {
-    if (this.open) this.afterOnHide();
     this.activator = params.activator;
-    if (this.open) this.afterOnShow();
-    if (this.open && !params.force) return false; // the show task will be forced to run so the next steps are unnecessary
+    if (this.open) {
+      this.afterOnHide();
```
Comment (dshovchko) [inline src/modules/esl-popup/core/esl-popup.ts:179]: Is it safe to unsubscribe in the `afterOnHide()` method when the activator has already changed?

Comment (ala-n) [reply src/modules/esl-popup/core/esl-popup.ts:179]: There is no longer required, ESL Event listeners knows exactly what was their targets on subscription

**Review by dshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Review by Natalie-Smirnova — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Pull request merged**